lib/msal-browser/src/interaction_client/BaseInteractionClient.ts

```diff
@@ -128,6 +132,18 @@ export abstract class BaseInteractionClient {
         return UrlString.getAbsoluteUrl(redirectUri, BrowserUtils.getCurrentUri());
     }
 
+    /*
+     * If authority provided in the request does not match environment/authority specified 
+     * in the account or MSAL config, we throw an error.
+     */
+    async validateRequestAuthority(authority: string, account: AccountInfo): Promise<void> {
+        const discoveredAuthority = await this.getDiscoveredAuthority(authority);
```

Comment:
This is an additional round trip. We need to leverage it to reuse it for the following AT call. However we do not need this information for native calls right? I want to review the end to end, make sure we do not tank our perf.

Reply:
The network call is only made if it hasn't already been done previously, so no it's not an extra round-trip. And, no we don't need to do it for native calls, but we may need to do it for any cache lookups

Reply:
Yeah that makes more sense. Thanks, resolving.